pandas/io/excel.py

```diff
@@ -270,6 +277,13 @@ def parse(self, sheetname=0, header=0, skiprows=None, skip_footer=0,
         if skipfooter is not None:
             skip_footer = skipfooter
 
+        if has_index_names:
```

Comment:
here check if its ``not None``, then provide the warning either way. So if they are passing it they will know.